File: sdk/dotnet/Heartlight.Hip.Tests/SynapticKernelTests.cs
using Heartlight.Hip;

namespace Heartlight.Hip.Tests;

public class SynapticKernelTests
{
    [Xunit.Fact]
    public void ConformanceV1()
    {
        var state = new SynapticState(0.2, 0.0);
        SynapticInput[] inputs =
        [
            new(0.5, 0.25, 1.0),
            new(1.0, -0.5, 0.4),
            new(0.2, 0.8, -0.25),
        ];
        double[] weights = [0.2096, 0.1967808, 0.2001622384];
        double[] traces = [0.125, -0.3875, -0.18875];
        for (var i = 0; i < inputs.Length; i++)
        {
            state = SynapticKernel.Step(state, inputs[i]);
            Xunit.Assert.InRange(Math.Abs(state.Weight - weights[i]), 0.0, 1e-12);
            Xunit.Assert.InRange(Math.Abs(state.Eligibility - traces[i]), 0.0, 1e-12);
        }
    }
}

File: sdk/swift/Tests/HeartlightHIPTests/SynapticKernelTests.swift
import XCTest
@testable import HeartlightHIP

final class SynapticKernelTests: XCTestCase {
    func testConformanceV1() throws {
        var state = SynapticState(weight: 0.2, eligibility: 0.0)
        let inputs = [
            SynapticInput(pre: 0.5, post: 0.25, reward: 1.0),
            SynapticInput(pre: 1.0, post: -0.5, reward: 0.4),
            SynapticInput(pre: 0.2, post: 0.8, reward: -0.25),
        ]
        let weights = [0.2096, 0.1967808, 0.2001622384]
        let traces = [0.125, -0.3875, -0.18875]
        for index in inputs.indices {
            state = try synapticStep(state, input: inputs[index])
            XCTAssertEqual(state.weight, weights[index], accuracy: 1e-12)
            XCTAssertEqual(state.eligibility, traces[index], accuracy: 1e-12)
        }
    }
}
